core/src/mindustry/logic/LExecutor.java

```diff
@@ -512,10 +512,19 @@ public void run(LExecutor exec){
                             exec.setobj(p4, null);
                         }else{
                             Tile tile = world.tileWorld(x1, y1);
-                            //any environmental solid block is returned as StoneWall, aka "@solid"
-                            Block block = tile == null ? null : !tile.synthetic() ? (tile.solid() ? Blocks.stoneWall : Blocks.air) : tile.block();
-                            exec.setobj(p3, block);
-                            exec.setobj(p4, tile != null && tile.build != null ? tile.build : null);
+                            if(tile == null){
+                                exec.setobj(p3, null);
+                            }else{
+                                if(exec.bool(p5)){
```

Comment:
This is very confusing; documented or not, most people would expect the 'floor' parameter to actually contain the floor instead of a boolean that is read.

Reply:
How do you feel about changing the 'floor' parameter to 'tryFloor' or even 'isFloor'?

Reply:
Why not just make it `floor` and write into it?

Reply:
Sometimes, I am dumb lmao